projects/igniteui-angular/src/lib/tree/tree-node/tree-node.component.ts

```diff
@@ -0,0 +1,695 @@
+     *      {{ node.label }}
+     *  </igx-tree-node>
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * const expanded = node.expanded;
+     * node.expanded = true;
+     * ```
+     */
+    @Input()
+    public get expanded() {
+        return this.treeService.isExpanded(this);
+    }
+
+    public set expanded(val: boolean) {
+        if (val) {
+            this.treeService.expand(this, false);
+        } else {
+            this.treeService.collapse(this);
+        }
+    }
+
+    /** @hidden @internal */
+    public get expandIndicatorTemplate(): TemplateRef<any> {
+        return this.tree?.expandIndicator ? this.tree.expandIndicator : this._defaultExpandIndicatorTemplate;
+    }
+
+    /**
+     * The native DOM element representing the node. Could be null in certain environments.
+     *
+     * ```typescript
+     * // get the nativeElement of the second node
+     * const node: IgxTreeNode = this.tree.nodes.first();
+     * const nodeElement: HTMLElement = node.nativeElement;
+     * ```
+     */
+    /** @hidden @internal */
+    public get nativeElement() {
+        return this.element.nativeElement;
+    }
+
+    /** @hidden @internal */
+    public ngOnInit() {
+        this.openAnimationDone.pipe(takeUntil(this.destroy$)).subscribe(
+            () => {
+                this.tree.nodeExpanded.emit({ owner: this.tree, node: this });
+            }
+        );
+        this.closeAnimationDone.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            this.tree.nodeCollapsed.emit({ owner: this.tree, node: this });
+            this.treeService.collapse(this);
+            this.cdr.markForCheck();
+        });
+    }
+
+    /** @hidden @internal */
+    public ngAfterViewInit() { }
+
+    /**
+     * @hidden @internal
+     * Sets the focus to the node's <a> child, if present
+     * Sets the node as the tree service's focusedNode
+     * Marks the node as the current active element
+     */
+    public handleFocus(): void {
+        if (this.disabled) {
+            return;
+        }
+        if (this.navService.focusedNode !== this) {
+            this.navService.focusedNode = this;
+        }
+        this.isFocused = true;
+        if (this.linkChildren?.length) {
+            this.linkChildren.first.nativeElement.focus();
+            return;
+        }
+        if (this.registeredChildren.length) {
+            this.registeredChildren[0].elementRef.nativeElement.focus();
+            return;
+        }
+    }
+
+    /**
+     * @hidden @internal
+     * Clear the node's focused status
+     */
+    public clearFocus(): void {
+        this.isFocused = false;
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public onSelectorClick(event) {
+        // event.stopPropagation();
+        event.preventDefault();
+        // this.navService.handleFocusedAndActiveNode(this);
+        if (event.shiftKey) {
```

Comment:
with this implementation if `shift` is clicked:
- active node changes which IMO is not correct
- you can only select range of nodes and cannot deselect range of nodes

Reply:
It still should change the active node, as, per spec, click change it.
For the range deselection - we were going for an implementation similar to how the tree grid behaved and there is not 'deselect range' behavior defined there (as the UX would be a bit clunky). We've exposed a "deselectAll" method, which developers can use to implement a deselect interaction in a specific app scenario.